integration/examples/lifecycle-hooks/skaffold.yaml

```diff
@@ -0,0 +1,16 @@
+apiVersion: skaffold/v2beta19
+kind: Config
+build:
+  artifacts:
+  - image: hooks-example
+    hooks:
+      before:
+        - command: ["sh", "-c", "./hook.sh"]
+          os: [darwin, linux]
+      after:
+        - command: ["sh", "-c", "docker images $SKAFFOLD_IMAGE --digests"]
```

Comment:
I wonder if there's a way for our yaml loading to handle this either as a string or array of strings.  Then we could avoid the `sh -c` or `command /c`, though I suppose we'd still have the `$XXX` vs `%XXX%`.

Reply:
ya the commands don't always carry over, which necessitated being able to specify the OS. With that, it's also left to the user to choose the shell.

Reply:
could we have some sort of default "invocation command" per OS that users could override? for `darwin`/`linux` we could use `sh -c` and for windows we could use `cmd.exe /C`.

Reply:
We can't get around the environment references though, and I *think* those will be pretty key.

We can look into supporting stringOrArray later, if we don't have support already.  If it's a string, then we feed it directly into `/bin/sh -c` or `cmd /c` accordingly.